src/libs/ReportActionsUtils.ts

```diff
@@ -635,6 +665,138 @@ function isNotifiableReportAction(reportAction: OnyxEntry<ReportAction>): boolea
     return actions.includes(reportAction.actionName);
 }
 
+function getMemberChangeMessageElements(reportAction: OnyxEntry<ReportAction>): MemberChangeMessageElement[] {
+    const isInviteAction = isInviteMemberAction(reportAction);
+    const verb = isInviteAction ? Localize.translateLocal('workspace.invite.invited') : Localize.translateLocal('workspace.invite.removed');
+
+    const messageElements: MemberChangeMessageElement[] = [
+        {
+            kind: 'text',
+            content: `${verb} `,
+        },
+    ];
+
+    const originalMessage = reportAction?.originalMessage as ChangeLog;
+    const targetAccountIDs: number[] = originalMessage?.targetAccountIDs ?? [];
+    const personalDetails = PersonalDetailsUtils.getPersonalDetailsByIDs(targetAccountIDs, 0);
+
+    const mentions = targetAccountIDs.map((accountID) => {
+        const personalDetail = personalDetails.find((personal) => personal.accountID === accountID);
+
+        if (personalDetail) {
+            const displayNameOrLogin =
+                LocalePhoneNumber.formatPhoneNumber(personalDetail.login ?? '') || (personalDetail?.displayName ? personalDetail?.displayName : Localize.translateLocal('common.hidden'));
+            return {content: `@${displayNameOrLogin}`, accountID};
+        }
+        return {content: `@${Localize.translateLocal('common.hidden')}`, accountID};
```

Comment:
If we extracted...

```ts
function getEffectiveDisplayName(personalDetails): string | undefined {
  if (personalDetail) {
    return LocalePhoneNumber.formatPhoneNumber(personalDetail.login ?? '') ?? personalDetail.displayName;
  }

  return undefined;
}
```

...we could use it like this:

```ts
return {content: `@${getEffectiveDisplayName(personalDetails) ?? Localize.translateLocal('common.hidden')}`, accountID};
```

...which reduces code duplication slightly.

Reply:
But, looking at this code, I wonder: should we prefer the phone number over user-configured present display name? 🤔